eth2util/eip712/eip712.go

```diff
@@ -64,7 +65,14 @@ func domainToType(domain Domain) Type {
 
 // HashTypedData returns the hash of the typed data.
 func HashTypedData(data TypedData) ([]byte, error) {
-	domainHash, err := hashData(domainToType(data.Domain))
+	domainType := domainToType(data.Domain)
+
+	// TODO(diogo): any better ideas?
```

Comment:
Having the type names as consts is slightly neater.

I don't get why are we deleting those fields?

Reply:
Weird. Probably add a comment to that function and explain the reasoning... And check with API team why is that.